### PR TITLE
Added modifications to polls and stories API

### PR DESCRIPTION
**Pull request by alviriseup**
Added modifications to polls and stories API call to solve the following two issues: #850 and #835

### CODE REVIEW

File: ureport/api/serializers.py [hunk 1]
```diff
@@ -100,6 +100,32 @@ def get_images(self, obj):
             for image in obj.get_featured_images()
         ]
 
+    # Function to use ?fields and ?exclude API calls for specific attributes in stories
+    # this is for which we can exclude any field
+    def __init__(self, *args, **kwargs):
+        # Don't pass the 'fields' arg up to the superclass
+        request = kwargs.get('context', {}).get('request')
+        str_fields = request.GET.get('exclude', '') if request else None
+        str_exclude_fields = request.GET.get('fields', '') if request else None
```
Comment (norkans7) [inline ureport/api/serializers.py:109]: Are not the logic inverted with the parameters used?

**Review by norkans7 — CHANGES_REQUESTED** [review]
Add tests

File: ureport/api/serializers.py (hunk 1 above)
Comment (alviriseup) [reply ureport/api/serializers.py]: You are correct. The naming convention was reversed; it is now fixed.

File: ureport/api/tests.py [hunk 2]
```diff
@@ -396,6 +396,32 @@ def test_polls_by_org_list_with_flow_uuid_parameter(self):
         self.assertEqual(response.data["count"], 1)
         self.assertEqual(response.data["results"][0]["title"], "registration")
 
+    def test_polls_by_org_list_with_fields_parameter(self):
+        url = "/api/v1/polls/org/%d/?fields=%s" % (self.uganda.pk, "title")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        count_polls = Poll.objects.filter(org=self.uganda, is_active=True, has_synced=True).count()
+        self.assertEqual(response.data["count"], count_polls)
+        poll = [self.second_featured_poll, self.first_featured_poll, self.another_poll, self.reg_poll]
```
Comment (norkans7) [inline ureport/api/tests.py:405]: variable name `polls` instead of poll as it is a list

File: ureport/api/serializers.py [hunk 3]
```diff
@@ -122,6 +147,31 @@ class PollReadSerializer(serializers.ModelSerializer):
...
+        fields = str_fields.split(',') if str_fields else None
+        exclude_fields = str_exclude_fields.split(',') if str_exclude_fields else None
+
+        # Instantiate the superclass normally
+        super(PollReadSerializer, self).__init__(*args, **kwargs)
+
+        if exclude_fields is not None:
+            # Drop any fields that are specified in the `exclude` argument.
+            exclude_allowed = set(exclude_fields)
+            print('exclude_allowed :',exclude_allowed)
```
Comment (norkans7) [inline ureport/api/serializers.py:165]: remove the print statement

File: ureport/api/tests.py (hunk 2 above)
Comment (alviriseup) [reply ureport/api/tests.py]: variable name has been changed from poll to `polls`

File: ureport/api/serializers.py (hunk 3 above)
Comment (alviriseup) [reply ureport/api/serializers.py]: the print statement has been removed.

**Review by norkans7 — CHANGES_REQUESTED** [review]
Make sure to run  the script `./code_check.py`

**Review by norkans7 — APPROVED** [review]
Thanks Alvi

**Pull request merged**